### PR TITLE
Add a Dockerfile

### PR DESCRIPTION
**Pull request by olix0r**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.git
+Dockerfile*
+target
+.cargo
```
Comment (hawkw) [inline .dockerignore:4]: Let's add `.github` as well, when #4 merges?

Comment (hawkw) [reply .dockerignore:4]: Also, should we ignore the `README`? It's not big, but it's an extra thing to copy into the docker image, and we don't want to invalidate the cache when it changes?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+ARG RUST_IMAGE=rust:1.42.0-buster
+ARG RUNTIME_IMAGE=debian:buster-slim
+
+FROM $RUST_IMAGE as build
+RUN rustup component add rustfmt
```
Comment (hawkw) [inline Dockerfile:5]: Why do we need rustfmt to build a docker image?

**Review by hawkw — COMMENTED** [review]
A couple of questions. Otherwise, looks good.

File: Dockerfile (hunk 2 above)
Comment (olix0r) [reply Dockerfile:5]: tonic

Comment (hawkw) [reply Dockerfile:5]: oh. right.

Comment (hawkw) [reply Dockerfile:5]: looks like we can also just [turn off the `rustfmt` thing](https://docs.rs/tonic-build/0.2.0/tonic_build/struct.Builder.html#method.format) — would that be better?

**Review by hawkw — APPROVED** [review]

**Review by kleimkuhler — APPROVED** [review]

**Pull request merged**